lib/plugins/aws/package/compile/events/apiGateway/lib/requestValidator.js

```diff
@@ -0,0 +1,216 @@
+'use strict';
+
+const _ = require('lodash');
+const BbPromise = require('bluebird');
+
+module.exports = {
+  compileRequestValidators() {
+    const apiGateway = this.serverless.service.provider.apiGateway || {};
+    const models = {};
+    // Models are defined in the provider and can be used in http events later
+    if (apiGateway.requestSchemas) {
+      for (const [schemaId, schemaConfig] of _.entries(apiGateway.requestSchemas)) {
+        let modelName;
+        let validatorName;
+        let description;
+        let definition;
+
+        // If schema is not defined this will try to map resourceDefinition as the schema
+        if (!schemaConfig.schema) {
+          definition = schemaConfig;
+        } else {
+          definition = schemaConfig.schema;
+        }
+
+        const resourceName = _.upperFirst(_.camelCase(schemaId));
+        const modelLogicalId = this.provider.naming.getModelLogicalId(
+          resourceName,
+          'any',
+          'application/json'
+        );
+
+        const validatorLogicalId = this.provider.naming.getValidatorLogicalId(resourceName, 'any');
+
+        if (schemaConfig.name) {
+          modelName = schemaConfig.name;
+          validatorName = `${modelName}Validator`;
+        }
+
+        if (schemaConfig.description) {
+          description = schemaConfig.description;
+        }
+
+        models[schemaId] = {
+          modelLogicalId,
+          validatorLogicalId,
+        };
+
+        _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources, {
+          [modelLogicalId]: {
+            Type: 'AWS::ApiGateway::Model',
+            Properties: {
+              RestApiId: { Ref: this.apiGatewayRestApiLogicalId },
+              Schema: definition,
+              ContentType: 'application/json',
+            },
+          },
+          [validatorLogicalId]: {
+            Type: 'AWS::ApiGateway::RequestValidator',
+            Properties: {
+              RestApiId: { Ref: this.apiGatewayRestApiLogicalId },
+              ValidateRequestBody: true,
+              ValidateRequestParameters: true,
+            },
+          },
+        });
+
+        if (modelName != null) {
```

Comment:
I think we can write it as `if (modelName)`, at least I would treat eventual empty string as lack of value